treenode.go

```diff
@@ -381,9 +381,19 @@ func (n *TreeNodeInstance) dispatchChannel(msgSlice []*ProtocolMsg) error {
 		for _, msg := range msgSlice {
 			out := reflect.ValueOf(n.channels[mt])
 			m := n.reflectCreate(to.Elem(), msg)
-			log.Lvl4(n.Name(), "Dispatching msg type", mt, " to", to, " :", m.Field(1).Interface())
+			log.Lvl4(n.Name(), "Dispatching msg type", mt, " to", to, ":", m.Field(1).Interface())
 			if out.Len() < out.Cap() {
-				out.Send(m)
+				func() {
+					defer func() {
+						if r := recover(); r != nil {
+							if r.(error).Error() == "send on closed channel" {
+								return
+							}
+							panic(r)
+						}
```

Comment:
The commit message says:     Catch writes on closed channels
    
    Getting a panic from a write on this channel can happen
    in legitimate scenarios related to messages arriving during
    shutdown, or in testing failed node configurations. Mask it,
    because it does not hurt anything.

Is that enough? Or do you still have Qs?

Reply:
I'm trying to understand when this would happen. The only time when we write to the channel is a call to Stop, I can't think of a situation for when it'll happen. Will this also mask errors from other channels?

Reply:
Ok I see what it's doing now. Nothing writes to it so it blocks.